File: rules/readable/fixtures/pass-ts-each-row-is-its-own-reported-case/duration.ts
export function formatDuration(seconds: number): string {
  const hours = Math.floor(seconds / SECONDS_PER_HOUR);
  const minutes = Math.floor((seconds % SECONDS_PER_HOUR) / SECONDS_PER_MINUTE);
  const remainder = seconds % SECONDS_PER_MINUTE;

  const parts = [`${remainder}s`];
  if (hours > 0 || minutes > 0) {
    parts.unshift(`${minutes}m`);
  }
  if (hours > 0) {
    parts.unshift(`${hours}h`);
  }

  return parts.join(" ");
}

const SECONDS_PER_MINUTE = 60;
const SECONDS_PER_HOUR = 3600;

File: rules/no-redundancy/fixtures/fail-go-duplicate-table-row/duration.go
package scenario

import "fmt"

func FormatDuration(seconds int) string {
	if seconds < secondsPerMinute {
		return fmt.Sprintf("%ds", seconds)
	}
	if seconds < secondsPerHour {
		return fmt.Sprintf("%dm %ds", seconds/secondsPerMinute, seconds%secondsPerMinute)
	}
	return fmt.Sprintf("%dh %dm", seconds/secondsPerHour, (seconds%secondsPerHour)/secondsPerMinute)
}

const (
	secondsPerMinute = 60
	secondsPerHour   = 3600
)
